### ISSUE TITLE
`Flightless Demo` loading fail

### ISSUE THREAD

**Issue by yangyangdaji** [ruffle-rs/ruffle#14150]
### `Flightless Demo` loading fail

![Flightless-Demo](https://github.com/ruffle-rs/ruffle/assets/108261729/0f08ef9b-57a6-4d0b-bf66-e4aacc46f6b7)

### Expected behavior

![ok](https://github.com/ruffle-rs/ruffle/assets/108261729/1ce42fc5-ed37-453d-8d93-b837c1bd4a84)

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0

**Comment by yangyangdaji** [comment]
# log
```shell
demo/:87 [Intervention]Slow network is detected. See https://www.chromestatus.com/feature/5636954674692096 for more details. Fallback font will be used while loading: https://ruffle.rs/demo/be36596da218e1eec01c.woff2
ruffle-player.js:488 New Ruffle instance created (Version: nightly 2023-11-25 | WebAssembly extensions: ON | Used renderer: wgpu-webgl)
ruffle-player.js:646 Loading SWF file logo-anim.swf
ruffle-player.js:570 Ruffle instance destroyed.
ruffle-player.js:488 New Ruffle instance created (Version: nightly 2023-11-25 | WebAssembly extensions: ON | Used renderer: wgpu-webgl)
ruffle-player.js:651 Loading SWF data
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.system.Security.allowDomain()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899281216' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899281216 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.URLLoader.close()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.close()
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.unloadAndStop()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.LocalConnection.connect()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899281217' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899281217 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899281218' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899281218 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185

```

**Comment by yangyangdaji** [comment]
# log
```shell
ruffle-player.js:570 Ruffle instance destroyed.
ruffle-player.js:488 New Ruffle instance created (Version: nightly 2023-11-25 | WebAssembly extensions: ON | Used renderer: wgpu-webgl)
ruffle-player.js:651 Loading SWF data
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.system.Security.allowDomain()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899803358' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899803358 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.URLLoader.close()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.close()
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.unloadAndStop()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.LocalConnection.connect()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803359' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803359 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803360' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803360 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
```

**Comment by yangyangdaji** [comment]
# Debug log
```shell
ruffle-player.js:570 Ruffle instance destroyed.
ruffle-player.js:488 New Ruffle instance created (Version: nightly 2023-11-25 | WebAssembly extensions: ON | Used renderer: wgpu-webgl)
ruffle-player.js:651 Loading SWF data
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/tag_utils.rs:477 Unknown tag code: 255
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.system.Security.allowDomain()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899803358' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/cachebusters/flightless.txt?gsCacheBusterID=1700899803358 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.URLLoader.close()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
requestAnimationFrame(异步)
(anonymous) @ ruffle_web-wasm_extensions.js:1308
M @ ruffle_web-wasm_extensions.js:235
V.n.wbg.__wbg_requestAnimationFrame_244e975fcad8e9a1 @ ruffle_web-wasm_extensions.js:1307
$ruffle_web::Ruffle::tick::h16f2669bf1186cf3 @ 0336c1d6:0x2d4d62
$wasm_bindgen_futures::future_to_promise::{{closure}}::{{closure}}::h6051c4a67062bc12 @ 0336c1d6:0x550b23
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.close()
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.display.Loader.unloadAndStop()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png net::ERR_FAILED
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::movie_loader::{{closure}}::h1531a59fbd4387f9 @ 0336c1d6:0x27eee5
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
ruffle_web-wasm_extensions.js:753 ERROR core/src/loader.rs:908 Error during movie loading of "https://s3.amazonaws.com/us_nitrome_s3/thumbnails/ico_flightless.png": FetchError("Got JS error")
ruffle_web-wasm_extensions.js:753 WARN core/src/stub.rs:126 Encountered stub: AVM2 flash.net.LocalConnection.connect()
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803359' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803359 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
demo/:1  Access to fetch at 'https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803360' from origin 'https://ruffle.rs' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
ruffle_web-wasm_extensions.js:1312 
        
        
        GET https://s3.amazonaws.com/us_nitrome_s3/games/flightless.pkg?gsCacheBusterID=1700899803360 net::ERR_FAILED 200 (OK)
V.n.wbg.__wbg_fetch_8cebc656dc6b11b1 @ ruffle_web-wasm_extensions.js:1312
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::fetch::{{closure}}::hfd63e3f578cf8d34 @ 0336c1d6:0x29e3b1
$ruffle_core::loader::Loader::load_url_loader::{{closure}}::he506bf211e012141 @ 0336c1d6:0x5f2d11
$<ruffle_web::navigator::WebNavigatorBackend as ruffle_core::backend::navigator::NavigatorBackend>::spawn_future::{{closure}}::h0f3c1a01989e7be1 @ 0336c1d6:0x410fd0
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::hc37e26db6daee956 @ 0336c1d6:0x72d858
$wasm_bindgen::convert::closures::invoke1_mut::h2c7feb67e27e88e8 @ 0336c1d6:0x9cacd2
$closure4444 externref shim @ 0336c1d6:0x9d6ea1
E @ ruffle_web-wasm_extensions.js:224
c @ ruffle_web-wasm_extensions.js:185
```

**Comment by mtv129** [comment]
That's an easy answer, you need to connect to a server from Nitrome, here is an example application:
https://www.newgrounds.com/projects/games/5219367/preview
the server exists, everything runs on as3 and it can also interact with the server, but it is impossible to receive data from it. and that's why it doesn't work

**Comment by yangyangdaji** [comment]
> That's an easy answer, you need to connect to a server from Nitrome

I know this, how to fix this problem 🤔

**Comment by yangyangdaji** [comment]
> That's an easy answer, you need to connect to a server from Nitrome
> the server exists, everything runs on as3 and it can also interact with the server, but it is impossible to receive data from it. and that's why it doesn't work

I know this, how to fix this problem 🤔

**Comment by yangyangdaji** [comment]
> That's an easy answer, you need to connect to a server from Nitrome
> the server exists, everything runs on as3 and it can also interact with the server, but it is impossible to receive data from it. and that's why it doesn't work

I know this, how to fix this problem 🤔

Does @Lord-McSweeney @sleepycatcoding have a suggestion for that?

**Comment by matu6968** [comment]
It does not seem to be a problem loading it on the desktop client using version [nightly-2025-02-28](https://github.com/ruffle-rs/ruffle/releases/tag/nightly-2025-02-28) but there are noticable glitches on the UI elements like buttons, which are not present in the official Flash player: 

![Image](https://github.com/user-attachments/assets/0ca1b4c8-7088-4f04-b870-2cc36f1efc7c)

![Image](https://github.com/user-attachments/assets/93536ff1-3783-4c25-8017-ba639f8007bb)

![Image](https://github.com/user-attachments/assets/1ad6d535-52ee-4d36-babf-7fbc365a146e)
![Image](https://github.com/user-attachments/assets/b66a7b13-74e4-4b43-8316-46f1c30a20a0)

`Debug logs:`
```
2025-02-28T17:55:57.378281Z  WARN sctk_adwaita::buttons: Ignoring unknown button type:     
2025-02-28T17:55:57.471909Z  INFO ruffle_desktop::gui::controller: Using preferred backend Vulkan
2025-02-28T17:55:57.525569Z  INFO ruffle_desktop::gui::controller: Using graphics API vulkan on NVIDIA GeForce GTX 1050 (type: DiscreteGpu)
2025-02-28T17:55:57.628033Z  WARN egui_wgpu::renderer: Detected a linear (sRGBA aware) framebuffer Bgra8UnormSrgb. egui prefers Rgba8Unorm or Bgra8Unorm    
2025-02-28T17:55:57.628813Z  INFO ruffle_desktop::gui::controller: loaded cjk fallback font "NotoSans-Regular"
2025-02-28T17:56:02.401579Z  WARN ruffle_video_external::decoder::openh264: Failed to load system OpenH264 library libopenh264.so.7: Error while loading OpenH264: libopenh264.so.7: cannot open shared object file: No such file or directory
2025-02-28T17:56:02.401659Z  WARN ruffle_video_external::decoder::openh264: Failed to load system OpenH264 library libopenh264.so.2.4.1: Error while loading OpenH264: libopenh264.so.2.4.1: cannot open shared object file: No such file or directory
2025-02-28T17:56:02.401718Z  WARN ruffle_video_external::decoder::openh264: Failed to load system OpenH264 library libopenh264.so: Error while loading OpenH264: libopenh264.so: cannot open shared object file: No such file or directory
2025-02-28T17:56:02.401726Z  INFO ruffle_video_external::decoder::openh264: Downloading OpenH264 library
2025-02-28T17:56:02.414257Z  INFO ruffle_video_external::decoder::openh264: Using OpenH264 at "/home/ximitekfun/.cache/ruffle/video/libopenh264-2.4.1-linux64.7.so"
2025-02-28T17:56:02.505593Z  INFO ruffle_core::context: Loaded SWF version 10, resolution 640x480 @ 30 FPS
2025-02-28T17:56:02.518001Z  WARN ruffle_core::tag_utils: Unknown tag code: 255
2025-02-28T17:56:02.526062Z  WARN ruffle_core::tag_utils: Unknown tag code: 255
2025-02-28T17:56:02.531652Z  INFO avm_trace: Nitrome Distribution Solution v1.05152012
2025-02-28T17:56:02.548695Z  WARN ruffle_core::stub: Encountered stub: AVM2 flash.system.Security.allowDomain()
2025-02-28T17:56:03.000500Z  WARN ruffle_core::stub: Encountered stub: AVM2 flash.display.Loader.load() addChild at the correct time
2025-02-28T17:56:03.000699Z  WARN ruffle_core::stub: Encountered stub: AVM2 flash.display.LoaderInfo.childAllowsParent getter
2025-02-28T17:56:03.003121Z  INFO avm_trace: Loading game...
2025-02-28T17:56:09.948428Z  WARN ruffle_core::tag_utils: Unknown tag code: 255
2025-02-28T17:56:10.623049Z  WARN ruffle_core::tag_utils: Unknown tag code: 255
2025-02-28T17:56:10.734585Z  WARN ruffle_core::stub: Encountered stub: AVM2 flash.display.DisplayObject.scale9Grid setter
2025-02-28T17:56:10.737556Z  INFO avm_trace: init queue length: 16
2025-02-28T17:56:10.737584Z  INFO avm_trace: percent per function: 6.25
2025-02-28T17:56:10.738916Z  WARN ruffle_core::stub: Encountered stub: AVM2 flash.net.SharedObject.close()
2025-02-28T17:56:16.054480Z  WARN symphonia_bundle_mp3::layer3: mpa: invalid main_data_begin, underflow by 91 bytes    
2025-02-28T17:56:18.887736Z  WARN symphonia_bundle_mp3::layer3: mpa: invalid main_data_begin, underflow by 91 bytes    
2025-02-28T17:56:19.554396Z  WARN symphonia_bundle_mp3::layer3: mpa: invalid main_data_begin, underflow by 91 bytes    
2025-02-28T17:56:24.048826Z  INFO ruffle_core::focus_tracker: Focus is now on Some(MovieClip(MovieClip { ptr: 0x561f7de05150 }))
2025-02-28T17:56:24.187289Z  INFO ruffle_core::focus_tracker: Focus is now on Some(Stage(Stage { ptr: 0x561f79dd3690 }))
2025-02-28T17:56:28.191457Z  INFO ruffle_core::focus_tracker: Focus is now on None
2025-02-28T17:56:34.744314Z  INFO ruffle_core::focus_tracker: Focus is now on Some(EditText(EditText { ptr: 0x561f7eca0ee0 }))
2025-02-28T17:56:36.634530Z  INFO ruffle_core::focus_tracker: Focus is now on None
2025-02-28T17:56:37.149323Z  INFO ruffle_core::focus_tracker: Focus is now on Some(MovieClip(MovieClip { ptr: 0x561f7ee40640 }))
2025-02-28T17:56:37.249576Z  INFO ruffle_core::focus_tracker: Focus is now on Some(Stage(Stage { ptr: 0x561f79dd3690 }))
2025-02-28T17:56:38.461334Z  INFO ruffle_core::focus_tracker: Focus is now on None
```
Regarding the problem of it not working on the web browser via the extension, it is a CORS issue by the browser logs, so a CORS proxy or a general CORS bypass would be needed: 

![Image](https://github.com/user-attachments/assets/6d94cf62-d6c6-432d-97b6-4bc486853716)

I have not tested this game fully, but so far there just could be UI related issues down the line.

**Comment by yangyangdaji** [comment]
@matu6968  Please create new issue
